root/healthcheck.sh

```diff
@@ -0,0 +1,9 @@
+#!/usr/bin/with-contenv bash
+set -euo pipefail
+IFS=$'\n\t'
+
+# universal check that the process is running
+pgrep -af nginx > /dev/null
```

Comment:
This is not really infallible due to how s6 overlay works. it restarts on a loop so you have a race condition with the pgrep and us trying to start nginx over and over again if it is busted

Reply:
To my knowledge the health check (even if it's reporting unhealthy) shouldn't be trying to restart the container at all. There are ways to do this, such as an external container that monitors the health status of other containers and restarts them if they are not healthy, but the health check in the container itself shouldn't be doing that.

I may be missing something but that's how I read the description in the documentation. From my experience running the official Plex image which has a health check it does not automatically restart when the status is not healthy.

Reply:
The health check is not doing anything, what I am saying is the health check has a very high likelyhood to return healthy even if the service is busted. S6 keeps starting the pid over an over again, so you have millisecond windows where pgrep is gonna return an OK in those start commands.

Reply:
Makes sense, and after testing with just this one I can see that you are correct. Would other containers all be the same? Or is this specific to nginx in some way? (I don't want to assume)

If checking that the process is running isn't going to help then maybe only one check per app makes sense.